torch/autograd/_functions/tensor.py

```diff
@@ -195,6 +201,29 @@ def backward(ctx, grad_output):
         return grad_tensor1, None, None, grad_tensor2, None
 
 
+class AdvancedIndexAdd(InplaceFunction):
+
+    @staticmethod
+    def forward(ctx, tensor1, adv_index, tensor2):
+        assert not ctx.needs_input_grad[1]
+        if ctx.needs_input_grad[2]:
+            ctx.adv_index = adv_index
+        ctx.mark_dirty(tensor1)
```

Comment:
i dont think you need to mark_dirty tensor1, as it's not an in-place or shared-storage operation (regular indexing is)